native/src/components/MapView.tsx

```diff
@@ -12,49 +14,84 @@ const MapContainer = styled.View`
 const StyledMap = styled(MapboxGL.MapView)`
   width: 700px;
   height: 500px;
+  z-index: 1;
 `
 
 type MapViewPropsType = {
   boundingBox: BBox
   featureCollection: FeatureCollection
+  currentFeature?: Feature<Point>
 }
 
 const textOffsetY = 1.25
-const layerProps: SymbolLayerProps = {
-  id: 'point',
-  style: {
-    symbolPlacement: 'point',
-    iconAllowOverlap: true,
-    iconIgnorePlacement: true,
-    iconImage: ['get', 'symbol'],
-    textField: ['get', 'title'],
-    textFont: ['Roboto Regular'],
-    textOffset: [0, textOffsetY],
-    textAnchor: 'top',
-    textSize: 12
-  }
-}
+const layerId = 'point'
 
 // Has to be set even if we use map libre
 MapboxGL.setAccessToken(mapConfig.accessToken)
-const MapView = ({ boundingBox, featureCollection }: MapViewPropsType): ReactElement => {
+const MapView = ({ boundingBox, featureCollection, currentFeature }: MapViewPropsType): ReactElement => {
+  const mapRef = React.useRef<MapboxGL.MapView | null>(null)
+  const cameraRef = React.useRef<MapboxGL.Camera | null>(null)
+  const [activeFeature, setActiveFeature] = useState<Feature<Point> | null>(currentFeature ?? null)
+  const [showPopup, setShowPopup] = useState<boolean>(!!currentFeature)
+  const layerProps: SymbolLayerProps = {
+    id: layerId,
+    style: {
+      symbolPlacement: 'point',
+      iconAllowOverlap: true,
+      iconIgnorePlacement: true,
+      iconImage: ['get', 'symbol'],
+      textField: ['case', ['==', ['get', 'id'], activeFeature?.properties?.id ?? -1], ['get', 'title'], ''],
+      textFont: ['Roboto Regular'],
+      textOffset: [0, textOffsetY],
+      textAnchor: 'top',
+      textSize: 12
+    }
+  }
+
   const bounds = {
     ne: [boundingBox[2], boundingBox[3]],
     sw: [boundingBox[0], boundingBox[1]]
   }
 
+  // if there is a current feature use the coordinates if not use bounding box
   const defaultSettings: CameraSettings = {
-    zoomLevel: defaultViewportConfig.zoom,
-    bounds
+    zoomLevel: activeFeature?.geometry.coordinates ? detailZoom : defaultViewportConfig.zoom,
+    centerCoordinate: activeFeature?.geometry.coordinates,
+    bounds: activeFeature?.geometry.coordinates ? undefined : bounds
   }
+
+  const onPress = useCallback(
+    async (e: Feature) => {
```

Comment:
```suggestion
    async (pressedLocation: Feature) => {
```
or a similar speaking name please